/**
 * This is a module containing control for accessories on the platform:
 * Buttons, lights, turn signals, etc.
 *
 */

syntax = "proto3";

/*
 * Car accessory module contains controls for generic car accessories
 */
package CarAccessoryModule;
option go_package = "../internal/pkg/ba_proto;ba_proto";
option csharp_namespace = "Google.Protobuf.ba_proto";


message ButtonStatus{
  bool isPressed = 1;
}

message ButtonCommand{
  enum Command{
    PRESS = 0;
  }
  Command command = 1;
}

message ButtonError{
  uint32 pressCount = 1;
}